src/tomato/ketchup/functions.py

```diff
@@ -57,20 +59,27 @@ def submit(args: Namespace) -> None:
     dirs = setlib.get_dirs(args.test)
     settings = setlib.get_settings(dirs.user_config_dir, dirs.user_data_dir)
     queue = settings["queue"]
-    if os.path.exists(args.payload) and os.path.isfile(args.payload):
-        log.debug(f"attempting to open Payload at '{args.payload}'")
-    else:
-        log.error(f"Payload file '{args.payload} not found.")
-        return None
-
-    with open(args.payload, "r") as infile:
-        if args.payload.endswith("json"):
-            pldict = json.load(infile)
-        elif args.payload.endswith("yml") or args.payload.endswith("yaml"):
-            pldict = yaml.full_load(infile)
+    if args.json:  # read directly from json string encoded in base64
+        log.debug("attempting to open payload delivered by base64 encoded json string")
+        encoded_json_string = args.payload
+        json_string = base64.b64decode(encoded_json_string).decode()
```

Comment:
Can probably combine these two:
```suggestion
        json_string = base64.b64decode(args.payload).decode()
```